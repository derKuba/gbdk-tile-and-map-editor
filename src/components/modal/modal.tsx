import { Component, Host, h, Prop } from '@stencil/core';

import Tunnel from '../../provider/StateProvider';

@Component({
    tag: 'mvjp-modal',
    styleUrl: 'modal.css',
    shadow: true,
})
export class Modal {
    /**
     *  display modal toggle event
     */
    @Prop() setIsModalOpen: Function;

    /**
     *  is modal open state
     */
    @Prop() isModalOpen: boolean;

    private closeModal = () => {
        this.setIsModalOpen(false);
    };

    render() {
        if (!this.isModalOpen) return null;

        return (
            <Host>
                <div id="myModal" class="modal">
                    <div class="modal-content">
                        <div class="close">
                            <button onClick={this.closeModal}> &times; </button>
                        </div>
                        <slot />
                    </div>
                </div>
            </Host>
        );
    }
}

Tunnel.injectProps(Modal, ['setIsModalOpen', 'isModalOpen']); // eslint-disable-line @stencil/ban-side-effects
