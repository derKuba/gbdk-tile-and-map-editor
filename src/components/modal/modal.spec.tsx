import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { Modal } from './modal';

describe('mvjp-modal', () => {
    it('builds', () => {
        expect(new Modal()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const page = await newSpecPage({
            components: [Modal],
            template: () => {
                return (
                    <div>
                        <mvjp-modal isModalOpen={false}></mvjp-modal>
                        <mvjp-modal isModalOpen={true}>
                            <div> i am content </div>
                        </mvjp-modal>
                    </div>
                );
            },
        });

        expect(page.root).toMatchSnapshot();
    });

    it('should close modal', async () => {
        const modal = new Modal();
        modal.isModalOpen = true;
        modal.setIsModalOpen = jest.fn();

        modal.closeModal();

        expect(modal.setIsModalOpen).toBeCalledWith(false);
    });
});
