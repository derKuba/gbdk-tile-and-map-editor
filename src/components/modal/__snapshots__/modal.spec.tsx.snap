// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-modal matches snapshot 1`] = `
<mvjp-modal>
  <mock:shadow-root></mock:shadow-root>
</mvjp-modal>
`;
